opendatahub-io/odh-dashboard
# Groups admin panel

# Groups admin panel

## Description
Added a new settings section to manage groups:

* Ability to control admin groups in the dashboard.
* Ability to control user groups in the dashbaord.
* Automatically detects a group selected is removed and notify the user.
* Adds the ability to support multi group selection.
*  Grant by default permission for accessing settings panel to all the **cluster admins**

## Kfdef

```yaml
kind: KfDef
apiVersion: kfdef.apps.kubeflow.org/v1
metadata:
  name: opendatahub
  namespace: opendatahub
spec:
  applications:
    - kustomizeConfig:
        repoRef:
          name: manifests
          path: odh-common
      name: odh-common
    - kustomizeConfig:
        parameters:
          - name: s3_endpoint_url
            value: s3.odh.com
        repoRef:
          name: manifests
          path: jupyterhub/jupyterhub
      name: jupyterhub
    - kustomizeConfig:
        overlays:
          - additional
        repoRef:
          name: manifests
          path: jupyterhub/notebook-images
      name: notebook-images
    - kustomizeConfig:
        overlays:
          - authentication
        repoRef:
          name: manifests
          path: odh-dashboard
      name: odh-dashboard
  repos:
    - name: kf-manifests
      uri: >-
        https://github.com/opendatahub-io/manifests/tarball/v1.4.0-rc.2-openshift
    - name: manifests
      uri: 'https://github.com/lucferbux/odh-manifests/tarball/groups-panel'
```

## How Has This Been Tested?



![test1](https://user-images.githubusercontent.com/16117276/178705990-bd01fbe4-5ffe-491b-87c9-e7b0d28c1b2f.gif)

1. Upload this KfDef.
2. Disable the operator to be able to modify the configmaps without being reconciled.
3. Check the instructions in [the docs](https://github.com/opendatahub-io/odh-dashboard/blob/main/docs/admin_dashboard.md) about how the admin dashboard is enabled.
4. Start adding and deleting groups.
5. When having a group selected, if it's deleted in the cluster, it will trigger a new notification.

## Merge criteria:



- [X] Github Issue: #192 
- [X] KfDef added.
